snap/snapenv/snapenv.go

```diff
@@ -95,6 +96,7 @@ func basicEnv(info *snap.Info) osutil.Environment {
 		// see https://github.com/snapcore/snapd/pull/2732#pullrequestreview-18827193
 		"SNAP_LIBRARY_PATH": "/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void",
 		"SNAP_REEXEC":       os.Getenv("SNAP_REEXEC"),
+		"SNAP_UID":          fmt.Sprint(sys.Geteuid()),
```

Comment:
why Geteuid? I would have expected Getuid?  fwiw bash defines both UID and EUID envvars

Reply:
Because it's what is used in other parts of the code:

![imagen](https://user-images.githubusercontent.com/105285003/222493648-0ed4bd90-b6b4-4c5a-8491-3b3269ed42d8.png)

Reply:
I see, I chased the code across refactorings, this was introduced with Geteuid by @jdstrand mentioning sudo as reason. I'm not entirely sure which behavior of sudo is relevant, maybe security can help reconstruct an argument for this so we can add a comment to the code and avoid future rehashing

Reply:
Regarding euid vs uid - if a process is running under sudo its effective uid would be 0 but its uid (ie real user id) could still be their unprivileged id. The effective id governs what permissions the kernel allows, so it may make more sense to make $SNAP_UID be the effective uid as this is what really matters. Also most users running something under sudo expect it to behave as though it was run directly by root and so using euid here seems more logical as that is the intention of the user.

However, I wonder if it would be better to just export both `SNAP_UID` and `SNAP_EUID` as `getuid()` and `geteuid()` respectively and let the snap itself respect whichever makes more sense to it.

Reply:
Added SNAP_UID and SNAP_EUID.